src/app/dataset/depublication/depublication.component.ts

```diff
@@ -0,0 +1,275 @@
+  selector: 'app-depublication',
+  templateUrl: './depublication.component.html',
+  styleUrls: ['./depublication.component.scss']
+})
+export class DepublicationComponent implements OnDestroy {
+  _datasetId: string;
+  currentPage = 0;
+  hasMore = false;
+  dataSortParam: SortParameter | undefined;
+  dataFilterParam: string | undefined;
+  depublicationData: Array<RecordPublicationInfo> = [];
+  depublicationSubscription: Subscription;
+  dialogFileOpen = false;
+  dialogInputOpen = false;
+  formRawText: FormGroup;
+  formFile: FormGroup;
+  isLoading = false;
+  isSaving = false;
+  optionsOpen = false;
+  pollingRefresh: Subject<boolean>;
+  sortHeaderGroupConf = {
+    cssClass: 'grid-header',
+    items: [
+      {
+        translateKey: 'depublicationColUrl',
+        fieldName: 'recordId'
+      },
+      {
+        translateKey: 'depublicationColStatus',
+        fieldName: 'depublicationStatus'
+      },
+      {
+        translateKey: 'depublicationColUnpublishedDate',
+        fieldName: 'depublicationDate'
+      }
+    ]
+  };
+
+  constructor(
+    private readonly depublications: DepublicationService,
+    private readonly errors: ErrorService,
+    private readonly fb: FormBuilder
+  ) {}
+
+  @Input()
+  set datasetId(id: string | undefined) {
+    if (id) {
+      this._datasetId = id;
+      this.buildForms();
+      this.beginPolling();
+    }
+  }
+
+  /** datasetId
+  /* getter for private variable (returns shadow variable)
+  */
+  get datasetId(): string | undefined {
+    return this._datasetId;
+  }
+
+  /** ngOnDestroy
+  /* unsubscribe from timer
+  */
+  ngOnDestroy(): void {
+    if (this.depublicationSubscription) {
+      this.depublicationSubscription.unsubscribe();
+    }
+  }
+
+  /** buildForm
+  /* - create a FormGroup
+  /* - update the form
+  */
+  buildForms(): void {
+    this.formRawText = this.fb.group({
+      recordIds: ['', [Validators.required, this.validateWhitespace]]
+    });
+    this.formFile = this.fb.group({
+      depublicationFile: ['', [Validators.required, this.validateFileExtension]]
+    });
+  }
+
+  /** validateWhitespace
+  /*  returns an error if the form control value is just whitespace
+  /*  @param {FormControl} control - the input control to validate
+  */
+  validateWhitespace(control: FormControl): { [key: string]: boolean } | null {
+    const val = control.value || '';
+    const isWhitespace = val.length > 0 && val.trim().length === 0;
+    return isWhitespace ? { whitespace: true } : null;
+  }
+
+  /** validateFileExtension
+  /*  returns an error if the form control value is the incorrect extension
+  /*  @param {FormControl} control - the input control to validate
+  */
+  validateFileExtension(control: FormControl): { [key: string]: boolean } | null {
+    const splitVal = (control.value ? control.value.name : '').split('.');
+    if (splitVal.length > 1 && splitVal[1].toLowerCase() !== 'txt') {
+      return { extension: true };
+    }
+    return null;
+  }
```

Comment:
Both these functions can be made static according to my IDE.

Reply:
See comment above

Reply:
In the general case: I understand as unit testing in Java has the same effect, preventing some functions from becoming static. Please ignore this comment.